react/features/recording/components/Recording/RecordButton.native.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+
+import { connect } from 'react-redux';
+
+import {
+    createRecordingDialogEvent,
+    sendAnalytics
+} from '../../../analytics';
+import { openDialog } from '../../../base/dialog';
+import { translate } from '../../../base/i18n';
+import { JitsiRecordingConstants } from '../../../base/lib-jitsi-meet';
+import {
+    PARTICIPANT_ROLE,
+    getLocalParticipant
+} from '../../../base/participants';
+import {
+    AbstractButton,
+    type AbstractButtonProps
+} from '../../../base/toolbox';
+
+import { getActiveSession } from '../../functions';
+
+import StartRecordingDialog from './StartRecordingDialog';
+import StopRecordingDialog from './StopRecordingDialog';
+
+/**
+ * The type of the React {@code Component} props of {@link RecordButton}.
+ */
+type Props = AbstractButtonProps & {
+
+    /**
+     * The current conference object.
+     */
+    _conference: Object,
+
+    /**
+     * True if there is a running active recording, false otherwise.
+     */
+    _isRecordingRunning: boolean,
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function,
+
+    /**
+     * The i18n translate function.
+     */
+    t: Function
+};
+
+/**
+ * An implementation of a button for starting and stopping recording.
+ */
+class RecordButton extends AbstractButton<Props, *> {
+    accessibilityLabel = 'Recording';
+    iconName = 'recEnable';
+    label = 'dialog.startRecording';
+    toggledIconName = 'recDisable';
+    toggledLabel = 'dialog.stopRecording';
+
+    /**
+     * Constructor of the component.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onStartRecording = this._onStartRecording.bind(this);
+    }
+
+    /**
+     * Handles clicking / pressing the button.
+     *
+     * @override
+     * @protected
+     * @returns {void}
+     */
+    _handleClick() {
+        const { _isRecordingRunning, dispatch } = this.props;
+
+        dispatch(openDialog(
+            _isRecordingRunning ? StopRecordingDialog : StartRecordingDialog
+        ));
+    }
+
+    /**
+     * Indicates whether this button is in toggled state or not.
+     *
+     * @override
+     * @protected
+     * @returns {boolean}
+     */
+    _isToggled() {
+        return this.props._isRecordingRunning;
+    }
+
+    _onStartRecording: () => void
+
+    /**
+     * Attempts to start the recording if the used confirmed the action.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onStartRecording() {
```

Comment:
Do you think it's too nested to have an abstract recording button? It might be.

Reply:
It seemed too nested for me (at least for now) as those buttons are handled totally differently on web (or on mobile, depends on who you ask) at the moment. But I'm happy to start a bigger refactor where we unify those codes too (overlay menu...etc)